src/esperanza/walletextension.h

```diff
@@ -202,6 +202,45 @@ class WalletExtension : public staking::StakingWallet {
       const finalization::VoteRecord &vote2);
 
   void PostInitProcess(CScheduler &scheduler);
+
+  template <typename Callable>
+  void ForEachMatureOutput(
+      const CTransaction &tx,
+      const CBlockIndex *block,
+      blockchain::Depth depth,
+      Callable f) const {
```

Comment:
I used function templates to work with callbacks. But I'm not sure this is the best approach.

Reply:
We (and bitcoin) use it in a lot of places, so I for one think this is fine. We have other (like `ForEackStakeableCoin`) functions in this unit that work like this and we have functions like that in `net` and `net_processing`.